File: cache/cache-core/src/test/resources/compat/issue-1369/src/NearJCacheConfigConsumer1121.java
package io.bluetape4k.cache.nearcache.jcache.compat.issue1369;

import io.bluetape4k.cache.nearcache.jcache.NearJCacheConfig;
import javax.cache.CacheManager;
import javax.cache.configuration.Factory;
import javax.cache.configuration.MutableConfiguration;

/** 1.12.1 Java consumer: public no-arg, direct five-arg constructor, and copy. */
final class NearJCacheConfigConsumer1121Java {

    private NearJCacheConfigConsumer1121Java() {
    }

    public static NearJCacheConfig<String, String> constructWithNoArguments() {
        return new NearJCacheConfig<>();
    }

    public static NearJCacheConfig<String, String> constructWithFiveArguments(
        Factory<CacheManager> cacheManagerFactory,
        String cacheName,
        MutableConfiguration<String, String> frontCacheConfiguration,
        boolean isSynchronous,
        long syncRemoteTimeout
    ) {
        return new NearJCacheConfig<>(
            cacheManagerFactory,
            cacheName,
            frontCacheConfiguration,
            isSynchronous,
            syncRemoteTimeout
        );
    }

    public static NearJCacheConfig<String, String> copyWithFiveArguments(
        NearJCacheConfig<String, String> source,
        Factory<CacheManager> cacheManagerFactory,
        String cacheName,
        MutableConfiguration<String, String> frontCacheConfiguration,
        boolean isSynchronous,
        long syncRemoteTimeout
    ) {
        return source.copy(
            cacheManagerFactory,
            cacheName,
            frontCacheConfiguration,
            isSynchronous,
            syncRemoteTimeout
        );
    }
}

File: cache/cache-core/src/test/resources/compat/issue-1369/src/NearJCacheConfigConsumer1121.kt
package io.bluetape4k.cache.nearcache.jcache.compat.issue1369

import io.bluetape4k.cache.nearcache.jcache.NearJCacheConfig

/** 1.12.1 Kotlin consumer: old five-field default dispatch and copy$default. */
object NearJCacheConfigConsumer1121 {
    @JvmStatic
    fun constructWithDefaults(): NearJCacheConfig<String, String> =
        NearJCacheConfig(cacheName = "legacy-kotlin")

    @JvmStatic
    fun copyWithDefaults(source: NearJCacheConfig<String, String>): NearJCacheConfig<String, String> =
        source.copy(cacheName = "legacy-kotlin-copy")
}
